Refratorization

vladbatalan opened the pull request.

vladbatalan committed:
Added SqlLite JDBC

vladbatalan committed:
Removed Map Reference from PVector

vladbatalan committed:
Some commnets added

invo12 commented on BatalanVlad/src/PaooGame/Physics/PVector.java:
```diff
@@ -1,46 +1,103 @@
+ * @author      Batalan Vlad
+ * @since 1.0
+ */
 
```
Class comments should be like this:
Class responsible for reprenting a 2D vector.

That's all, no explanations given

vladbatalan replied:
Done

invo12 commented on BatalanVlad/src/PaooGame/Physics/PVector.java:
```diff
@@ -1,46 +1,103 @@
+    /**
+     * The x coordonate of the PVector.
+     */
     private float x;
```
No comments needed for fields, just for public functions

vladbatalan replied:
Done

invo12 commented on BatalanVlad/src/PaooGame/Physics/PVector.java:
```diff
@@ -1,46 +1,103 @@
+    /**
+     * Default constructor. All the coordonates are initialised with 0.
+     */
     public PVector(){
```
Same ideea, no need to comment constructors, getters and setters, it's common sense

vladbatalan replied:
Solved

invo12 commented on BatalanVlad/src/PaooGame/Physics/PVector.java:
```diff
@@ -1,46 +1,103 @@
+     * @throws CloneNotSupportedException
+     *  If clonning is not supported
+     */
     @Override
```
Be short and on point. Maybe something like this:
Function responsible for creating a copy of the PVector.
Try to leave a newline between first comment and the rest.
Example:
"Text

@return..

vladbatalan replied:
Done

invo12 commented on BatalanVlad/src/PaooGame/Physics/PVector.java:
```diff
@@ -1,46 +1,103 @@
+ */
 
 public class PVector {
+
```
What does Pvector stands for, should refactor name to be easy to read. If P stands for point refactor something like PointVector, or something like this

vladbatalan replied:
Refactored to PointVector

invo12 commented on BatalanVlad/src/PaooGame/Tiles/Map.java:
```diff
@@ -119,7 +113,7 @@ public void Draw(Graphics g){
 
-    public void MapExtend(){
+    public void mapExtend(){
         if(map_height < MIN_HEIGHT){
```
Try to leave a blank line between function name and instructions, looks nicer. Also if you have for loops,if,while, try to leave a blank line asswell if the body has more than 1 line.

vladbatalan replied:
Tried to do this for every occasion

invo12 commented on BatalanVlad/src/PaooGame/Game.java:
```diff
@@ -15,26 +15,67 @@
+/**
+ * The main instance of the game
+ */
 public class Game implements Runnable
```
Class responsible for running an instance of the game. All class comments should start with "class responsible for"

vladbatalan replied:
I left this as it is and I'll try to use for the next ones

invo12 commented on BatalanVlad/src/PaooGame/Game.java:
```diff
@@ -15,26 +15,67 @@
+     * The main window of the game.
+     */
+    private GameWindow      wnd;
+
```
Make names suggestive, you have recommandations from IDE, use them, rename wnd to gameWindow. Make code easy to read

vladbatalan replied:
Refactored to gameWindow

invo12 commented on BatalanVlad/src/PaooGame/Game.java:
```diff
@@ -134,7 +170,7 @@ public synchronized void StartGame()
+        if(runState)
         {
                 /// Actualizare stare thread
             runState = false;
```
Don't over comment. If you need comments to explain something that is not a class/public function you have one of the following situations:
1) that piece of code is hard and it needs clarifications
2) your code is not expressive enough, and you need refactorization

If you are not in 1) then you don't need to comment that stuff. Keep that in mind.

vladbatalan replied:
Made the simplifications

invo12 commented on BatalanVlad/src/PaooGame/Physics/Body.java:
```diff
@@ -127,21 +131,18 @@ public void ajustPositionOnCollision(){
 
     public void Stand(){
         //if only one command is running or none
-        if(actions[0] == false && actions[1] == false)
```
Okay, so if ever, the order of actions change, your game is done for life. Use an enum better, or a mapping function between the array and the enum. But this is not necesarry for this commit, but next ones.

vladbatalan committed:
Name refractoring

vladbatalan committed:
Database singleton update

Every operation in the databbase now creates a connection and destroys it at the end of the operation.
